enterprise/cli/server_dbcrypt_test.go

```diff
@@ -191,30 +216,84 @@ func TestServerDBCrypt(t *testing.T) {
 	}
 }
 
-func genData(t *testing.T, db database.Store, n int) []database.User {
+func genData(t *testing.T, db database.Store) []database.User {
 	t.Helper()
 	var users []database.User
-	for i := 0; i < n; i++ {
-		usr := dbgen.User(t, db, database.User{
-			LoginType: database.LoginTypeOIDC,
-		})
-		_ = dbgen.UserLink(t, db, database.UserLink{
-			UserID:            usr.ID,
-			LoginType:         usr.LoginType,
-			OAuthAccessToken:  "access-" + usr.ID.String(),
-			OAuthRefreshToken: "refresh-" + usr.ID.String(),
-		})
-		_ = dbgen.GitAuthLink(t, db, database.GitAuthLink{
-			UserID:            usr.ID,
-			ProviderID:        "fake",
-			OAuthAccessToken:  "access-" + usr.ID.String(),
-			OAuthRefreshToken: "refresh-" + usr.ID.String(),
-		})
-		users = append(users, usr)
+	// Make some users
+	for _, status := range database.AllUserStatusValues() {
+		for _, loginType := range database.AllLoginTypeValues() {
+			for _, deleted := range []bool{false, true} {
+				usr := dbgen.User(t, db, database.User{
+					LoginType: loginType,
+					Status:    status,
+					Deleted:   deleted,
+				})
+				_ = dbgen.GitAuthLink(t, db, database.GitAuthLink{
+					UserID:            usr.ID,
+					ProviderID:        "fake",
+					OAuthAccessToken:  "access-" + usr.ID.String(),
+					OAuthRefreshToken: "refresh-" + usr.ID.String(),
+				})
+				// Fun fact: our schema allows _all_ login types to have
+				// a user_link. Even though I'm not sure how it could occur
+				// in practice, making sure to test all combinations here.
+				_ = dbgen.UserLink(t, db, database.UserLink{
+					UserID:            usr.ID,
+					LoginType:         usr.LoginType,
+					OAuthAccessToken:  "access-" + usr.ID.String(),
+					OAuthRefreshToken: "refresh-" + usr.ID.String(),
+				})
+				users = append(users, usr)
+			}
+		}
 	}
 	return users
 }
 
+func dumpUsers(t *testing.T, db *sql.DB) {
```

Comment:
Should we limit the OS test env for this test?

Reply:
If there were the possibility of either leaking sensitive information or performing destructive database changes, I would say yes. I don't believe either of those are the case here, though.

Reply:
In any case, I'll be removing this in a follow-up PR and replacing it with a more general "pg_dump on test failure".